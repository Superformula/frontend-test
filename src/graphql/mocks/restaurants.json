{
  "search": {
    "__typename": "Businesses",
    "total": 5028,
    "business": [
      {
        "__typename": "Business",
        "id": "faPVqws-x-5k2CQKDNtHxw",
        "photos": [
          "https://s3-media4.fl.yelpcdn.com/bphoto/_zXRdYX4r1OBfF86xKMbDw/o.jpg"
        ],
        "name": "Yardbird Southern Table & Bar",
        "rating": 1.5,
        "categories": [
          {
            "__typename": "Category",
            "alias": "southern",
            "title": "Southern"
          },
          {
            "__typename": "Category",
            "alias": "newamerican",
            "title": "American (New)"
          },
          {
            "__typename": "Category",
            "alias": "cocktailbars",
            "title": "Cocktail Bars"
          }
        ],
        "price": "$$",
        "is_closed": true
      },
      {
        "__typename": "Business",
        "id": "4JNXUYY8wbaaDmk3BPzlWw",
        "photos": [
          "https://s3-media1.fl.yelpcdn.com/bphoto/tEcdu80k_iS4OIqEVSDkZw/o.jpg"
        ],
        "name": "Mon Ami Gabi",
        "rating": 4,
        "categories": [
          { "__typename": "Category", "alias": "french", "title": "French" },
          {
            "__typename": "Category",
            "alias": "steak",
            "title": "Steakhouses"
          },
          {
            "__typename": "Category",
            "alias": "breakfast_brunch",
            "title": "Breakfast & Brunch"
          }
        ],
        "price": "$$",
        "is_closed": true
      },
      {
        "__typename": "Business",
        "id": "vHz2RLtfUMVRPFmd7VBEHA",
        "photos": [
          "https://s3-media4.fl.yelpcdn.com/bphoto/mFimEc6u8JCmbM5X1qgCeQ/o.jpg"
        ],
        "name": "Gordon Ramsay Hell's Kitchen",
        "rating": 4.5,
        "categories": [
          {
            "__typename": "Category",
            "alias": "newamerican",
            "title": "American (New)"
          }
        ],
        "price": "$$$",
        "is_closed": false
      },
      {
        "__typename": "Business",
        "id": "hihud--QRriCYZw1zZvW4g",
        "photos": [
          "https://s3-media4.fl.yelpcdn.com/bphoto/SrM8q-cGpbX5RJUYiEFFrQ/o.jpg"
        ],
        "name": "Gangnam Asian BBQ Dining",
        "rating": 5,
        "categories": [
          {
            "__typename": "Category",
            "alias": "japanese",
            "title": "Japanese"
          },
          { "__typename": "Category", "alias": "korean", "title": "Korean" },
          { "__typename": "Category", "alias": "bbq", "title": "Barbeque" }
        ],
        "price": "$$",
        "is_closed": true
      },
      {
        "__typename": "Business",
        "id": "3kdSl5mo9dWC4clrQjEDGg",
        "photos": [
          "https://s3-media3.fl.yelpcdn.com/bphoto/tDnWv0HCj3YgRRIzLuMfww/o.jpg"
        ],
        "name": "Egg & I",
        "rating": 4.5,
        "categories": [
          {
            "__typename": "Category",
            "alias": "breakfast_brunch",
            "title": "Breakfast & Brunch"
          },
          { "__typename": "Category", "alias": "burgers", "title": "Burgers" },
          {
            "__typename": "Category",
            "alias": "tradamerican",
            "title": "American (Traditional)"
          }
        ],
        "price": "$$",
        "is_closed": false
      },
      {
        "__typename": "Business",
        "id": "JDZ6_yycNQFTpUZzLIKHUg",
        "photos": [
          "https://s3-media3.fl.yelpcdn.com/bphoto/C2rDffzQgMayXBDNknx7fQ/o.jpg"
        ],
        "name": "El Dorado Cantina - Las Vegas Strip",
        "rating": 4.5,
        "categories": [
          { "__typename": "Category", "alias": "mexican", "title": "Mexican" },
          { "__typename": "Category", "alias": "bars", "title": "Bars" },
          {
            "__typename": "Category",
            "alias": "latin",
            "title": "Latin American"
          }
        ],
        "price": "$$",
        "is_closed": false
      },
      {
        "__typename": "Business",
        "id": "QXV3L_QFGj8r6nWX2kS2hA",
        "photos": [
          "https://s3-media4.fl.yelpcdn.com/bphoto/pu9doqMplB5x5SEs8ikW6w/o.jpg"
        ],
        "name": "Nacho Daddy",
        "rating": 4.5,
        "categories": [
          {
            "__typename": "Category",
            "alias": "newamerican",
            "title": "American (New)"
          },
          { "__typename": "Category", "alias": "mexican", "title": "Mexican" },
          {
            "__typename": "Category",
            "alias": "breakfast_brunch",
            "title": "Breakfast & Brunch"
          }
        ],
        "price": "$$",
        "is_closed": false
      },
      {
        "__typename": "Business",
        "id": "mDR12Hafvr84ctpsV6YLag",
        "photos": [
          "https://s3-media1.fl.yelpcdn.com/bphoto/goAZephpY5xHQlkXWZFYvw/o.jpg"
        ],
        "name": "Mr. Mamas Breakfast and Lunch",
        "rating": 4.5,
        "categories": [
          {
            "__typename": "Category",
            "alias": "breakfast_brunch",
            "title": "Breakfast & Brunch"
          },
          {
            "__typename": "Category",
            "alias": "tradamerican",
            "title": "American (Traditional)"
          },
          { "__typename": "Category", "alias": "burgers", "title": "Burgers" }
        ],
        "price": "$",
        "is_closed": false
      }
    ]
  }
}
