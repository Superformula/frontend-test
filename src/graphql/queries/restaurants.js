import { gql } from '@apollo/client';

export const restaurantsQuery = gql`
  query GetRestaurants(
    $limit: Int = 12
    $offset: Int = 0
    $open_now: Boolean
    $price: String
    $categories: String
  ) {
    search(
      location: "Las Vegas"
      offset: $offset
      limit: $limit
      open_now: $open_now
      price: $price
      categories: $categories
    ) {
      total
      business {
        id
        name
        rating
        photos
        price
        is_closed
        categories {
          alias
          title
        }
      }
    }
  }
`;
