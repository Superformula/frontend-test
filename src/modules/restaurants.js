import { combineReducers } from 'redux';
import { createAction, handleActions } from 'redux-actions';

import * as constants from '../constants';
import request from '../helpers/request';

// Action Creators
const fetchRestaurantsRequest = createAction('RESTAURANTS_FETCH_REQUEST');
const fetchRestaurantsResponse = createAction('RESTAURANTS_FETCH_RESPONSE');

export const fetchRestaurants = params => (dispatch) => {
  const queryObject = { ...{ location: constants.DEFAULT_LOCATION, term: 'restaurants', limit: 8 }, ...params };
  const queryString = Object.keys(queryObject).map((filter) => {
    if (queryObject[filter]) {
      return `${filter}=${queryObject[filter]}`;
    }
    return null;
  }).filter(item => item).join('&');
  dispatch(fetchRestaurantsRequest());
  request({ url: `${constants.API_URL}businesses/search?${queryString}` })
    .then((response) => {
      dispatch(fetchRestaurantsResponse(response));
    })
    .catch((err) => {
      dispatch(fetchRestaurantsResponse(err));
    });
};

// Reducers
const loading = handleActions({
  [fetchRestaurantsRequest]() {
    return true;
  },
  [fetchRestaurantsResponse]() {
    return false;
  },
}, true);

const data = handleActions({
  [fetchRestaurantsResponse]: {
    next(state, { payload }) {
      if (state && state.businesses) {
        const { businesses } = state;
        return {
          ...state,
          businesses: [
            ...businesses,
            ...payload.businesses,
          ],
        };
      }
      return payload;
    },
  },
}, {});

const error = handleActions({
  [fetchRestaurantsResponse]: {
    next() {
      return null;
    },
    throw(state, { payload: { message } }) {
      return message;
    },
  },
}, null);

export default combineReducers({
  error,
  loading,
  data,
});

// Selectors
export const getRestaurants = state => state.restaurants.data;
export const getRestaurantsError = state => state.restaurants.error;
export const getRestaurantsLoading = state => state.restaurants.loading;
