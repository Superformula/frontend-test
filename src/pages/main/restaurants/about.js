import React from "react";
import styled from "styled-components";

const Title = styled.h1`
  font-family: ${props => props.theme.fontFamily};
  font-weight: ${props => props.theme.fontWeightLight};
  font-size: 54px;
  color: ${props => props.theme.colorHeading};
  margin: 0;
  padding: 0;
  margin-bottom: 24px;
`;

const Description = styled.p`
  font-family: ${props => props.theme.fontFamily};
  font-weight: ${props => props.theme.fontWeightLight};
  font-size: 22px;
  color: ${props => props.theme.colorText};
  margin: 0;
  padding: 0;
  margin-bottom: 36px;
`;

export default function About() {
  return (
    <section>
      <Title>Restaurants</Title>
      <Description>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
        tempor incididunt ut labore et dolore magna aliqua.
      </Description>
    </section>
  );
}
