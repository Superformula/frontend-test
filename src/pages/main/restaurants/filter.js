import styled from "styled-components";
import React from "react";

import RadioCheckbox from "../../../components/ui/checkbox_radio";
import Dropdown from "../../../components/ui/dropdown";

const Container = styled.section`
  padding-top: 26px;
  padding-bottom: 26px;
  margin-left: ${props => props.theme.bodyPaddingLeft};
  margin-right: ${props => props.theme.bodyPaddingRight};
`;

const FilterText = styled.span`
  display: inline-block;
  font-family: ${props => props.theme.fontFamily};
  font-size: ${props => props.theme.fontSizeFilter};
  color: ${props => props.theme.colorFilterBy};
  margin-right: ${props => props.theme.marginRightFilter};
`;

const OpenNow = styled.div`
  border-bottom: 1px solid #c8c8c8;
  margin: 0;
  padding: 0;
  padding-bottom: 6px;
  display: inline-block;
`;

const OpenNowText = styled.label`
  font-family: "Helvetica Neue", Helvetica, sans-serif;
  font-size: 16px;
  margin-left: 8px;
`;

const Price = styled.span`
  font-family: "Helvetica Neue", Helvetica, sans-serif;
  font-size: 16px;
`;

const Category = styled.select``;

export default function Filter() {
  return (
    <Container>
      <FilterText>Filter By:</FilterText>
    </Container>
  );
}

/*
<OpenNow>
          <RadioCheckbox id="open-now" />
          <OpenNowText htmlFor="open-now">Open Now</OpenNowText>
        </OpenNow>
        <Dropdown title={<Price>Price</Price>} width={"10em"} />
        */
