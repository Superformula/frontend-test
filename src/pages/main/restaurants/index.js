import React from "react";
import styled from "styled-components";

import About from "./about";

const Bar = styled.hr`
  color: #e6e6e6;
`;

export default function Resturants() {
  return (
    <main className="restaurants">
      <About />
      <Bar />
    </main>
  );
}
