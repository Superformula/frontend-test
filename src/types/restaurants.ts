export type RestaurantInfo = {
  id: string;
  name: string;
  photo: string;
  rating: number;
  price: string;
  cuisineName: string;
  isOpen: boolean;
  photoAlt?: string;
};

export type Review = {
  dateCreated: string;
  rating: number;
  comment: string;
  user: {
    name: string;
    imageUrl: string;
  };
};

export type RestaurantDetailedInfo = {
  name: string;
  rating: number;
  cuisineName: string;
  priceRange: string;
  isOpen: boolean;
  reviewsCount: number;
  images: string[];
  reviews: Review[];
};
