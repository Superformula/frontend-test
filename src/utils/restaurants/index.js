import { ROUTES } from 'consts/routes';

export const getRestaurantUrl = id => ROUTES.RESTAURANT.replace(':id', id);
