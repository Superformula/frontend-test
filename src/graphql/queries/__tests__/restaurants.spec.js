import { restaurantsQuery } from '../restaurants';

describe('restaurantsQuery', () => {
  it('should build correct restaurant query', () => {
    expect(restaurantsQuery).toMatchSnapshot();
  });
});
