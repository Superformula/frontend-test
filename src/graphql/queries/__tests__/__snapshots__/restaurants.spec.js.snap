// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restaurantsQuery should build correct restaurant query 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "GetRestaurants",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "location",
                },
                "value": Object {
                  "block": false,
                  "kind": "StringValue",
                  "value": "Las Vegas",
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "offset",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "offset",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "limit",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "limit",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "open_now",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "open_now",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "price",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "price",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "categories",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "categories",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "search",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "total",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "business",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "rating",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "photos",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "price",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "is_closed",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "categories",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "alias",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "title",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": Object {
            "kind": "IntValue",
            "value": "12",
          },
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "Int",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "limit",
            },
          },
        },
        Object {
          "defaultValue": Object {
            "kind": "IntValue",
            "value": "0",
          },
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "Int",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "offset",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "Boolean",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "open_now",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "price",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "categories",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 501,
    "start": 0,
  },
}
`;
